File: NancyWebBlog/Models/PostPreviewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using NancyWebBlog.Model;

namespace NancyWebBlog.Models
{
    public class PostPreviewModel
    {
        public PostPreviewModel(Post post)
        {
            ID = post.ID;
            Author = post.Author;
            LogoUrl = post.LogoUrl;
            PreBody = post.PreBody;
            PostedAt = post.PostedAt;
            Title = post.Title;
        }

        public int ID { get; set; }

        public string Author { get; set; }

        public string LogoUrl { get; set; }

        public string PreBody { get; set; }

        public DateTime PostedAt { get; set; }

        public string Title { get; set; }
    }
}

File: NancyWebBlog/Scripts/viewModel/PostPreviewModel.js
function PreviewPostModel(data) {
    var self = this

    self.title = data.Title;
    self.id = data.ID;
    self.author = data.Author;
    self.postedAt = new Date(Date.parse(data.PostedAt));
    self.preBody = data.PreBody;
}
